### DISCUSSION TITLE
Disabled webserver via webconsole, how to start it again by uart?

### DISCUSSION THREAD

**Discussion by yingpwu** [arendst/Tasmota#21768] [Q&A] [answered]
# Problem
 I runned command webserver 1 ,so it going to user mode , 
open http://ip/cs page ,cant open the console page ,
how to change the webserver to 2 mode? 
# Tried
I searched in command help docs ,but didn't figure it out, 
none of the commands took effect i sent,
then i capture the tasmota install page webconsole command ,
it use some  HEX code by" IMPROV xx xx xx xx " , and didn't found it in command help docs.
    How to send the uart command  correctly here?

**Comment by sfromis** [comment, marked as answer]
If you have a serial console via UART, you can just issue the normal `WebServer` command there.

Beware that it will only work in many terminal programs if you send the command lines by ctrl-J instead of the enter key. And your typing will only be visible if you config the terminal program for half duplex mode, and backspace is not supported. I generally prefer to have the command already, and copy/paste it into a serial console.

Not sure what you are even trying to with that IMPROV stuff, but it is unrelated to just issuing normal commands. No reason to "waste time" on capturing anything, unless you specifically want to explore how the install page works.

Or you can just issue commands via MQTT, which tends to be the easiest option.

**Reply by yingpwu** [reply]
thanks , it's working by using moba xterm , but I'm curious about why it didn't work in simple uart assistant tool .

**Reply by sfromis** [reply]
That depends on the complexities of that "simple" UART assistant. The word "assistant" suggests that it may be more complex than just a terminal program.

**Reply by yingpwu** [reply]
Maybe i see it, the uart assistant use escape chars "\n", so i manul add HEX 0a at the end of message i sent  instead of ctrl+J in terminal program, it working in assistant tools

**Reply by sfromis** [reply]
Well, that's related to why I reminded you to use ctrl-J to send lines, which is **exactly** the same as hex `0a`.

**Reply by yingpwu** [reply]
TYSM
